3rdparty/obiwarp/vec.cpp

```diff
@@ -359,95 +277,63 @@ VecF VecF::operator+(const VecF &A) {
         return blank;
     }
 
-    else {
-        VecF *C = new VecF(_n);
-        VecF tmp = *C;
-        tmp._to_pass_up = C; 
+else {
+    VecF *C = new VecF(_n);
+    VecF tmp = *C;
+    tmp._to_pass_up = C;
         printf("TMPENEW %d\n", tmp.shallow());
-        for (int i = 0; i < _n; ++i) {
-            tmp[i] = _dat[i] + A[i];
-        }
-        return tmp;
+    for (int i = 0; i < _n; ++i) {
+        tmp[i] = _dat[i] + A[i];
     }
+    return tmp;
+}
 }
 */
 
 float VecF::sum() {
-    float *me = (float*)(*this);
-    float sum = 0;
-    for( int n = 0; n < _n; n++) {
-        sum += me[n];
-    }
-    return sum;
+    return std::accumulate(_dat.begin(), _dat.end(), 0.0f);
 }
 
 char * VecF::class_name() {
-    char *name = new char[7]; 
+    char *name = new char[7];
     strcpy(name, "VecF");
     return name;
 }
 
-
 void VecF::abs_val() {
     for (int n = 0; n < _n; ++n) {
         if (_dat[n] < 0) { _dat[n] *= -1; }
     }
 }
 
-
 void VecF::std_normal() {
     // @TODO: would like avg and stdev to stay double, even for floats!
-    (*this) -= (float)this->avg();
+    (*this) -= static_cast<float>(this->avg());
     double mean, stdev;
     this->sample_stats(mean, stdev);
-    (*this) /= (float)stdev;
+    (*this) /= static_cast<float>(stdev);
 }
 
-
 void VecF::remove(int index) {
-    float *_tmp_arr = new float[_n - 1];
-    int _cnt = 0;
-    for (int i = 0; i < _n; ++i) {
-        if (i != index) {
-            _tmp_arr[_cnt] = _dat[i];
-            ++_cnt;
-        }
-    }
-    if (!_shallow) {
-        delete []_dat;
-    }
+    _dat.erase(_dat.begin() + index);
     _n = _n - 1;
-    _dat = _tmp_arr;
-    _shallow = false;
-}
-
-// Could be faster for ints
-int VecF::floatCompare( const void *a, const void *b ) {
-    float c = *(float *)a - *(float *)b;
-    if ( c < 0 ) return -1;
-    if ( c > 0 ) return 1;
-    return 0;
 }
 
 void VecF::sort() {
-    qsort(_dat, _n, sizeof(float), floatCompare);
+    std::sort(_dat.begin(), _dat.end());
 }
 
-int VecF::index(float val) { 
-    for (int i = 0; i < _n; ++i) {
-        if (val == _dat[i]) {
-            return i;
-        }
+int VecF::index(float val) {
+    auto result = std::find(_dat.begin(), _dat.end(), val);
+    if (result != _dat.end()) {
+        return (result - _dat.begin());
     }
     return -1;
 }
 
 double VecF::avg() const {
-    double total = 0;
-    for( int n = 0; n < _n; ++n) {
-        total += _dat[n];
-    }
-    return total/_n;
+    double total = std::accumulate(_dat.begin(), _dat.end(), 0.0);
+    return total / static_cast<double>(_n);
```

Comment:
Because I believe that when doing real number arithmetic, its always a good practice to make sure all values involved are (or are type-casted to be) of the same type (`double` in this case).